fe/fe-core/src/main/java/com/starrocks/http/rest/RestBaseAction.java

```diff
@@ -292,4 +301,81 @@ protected static int getPageSize(BaseRequest request) {
         });
     }
 
+    public static List<String> fetchResultFromOtherFrontendNodes(String queryPath,
+                                                                 String authorization,
+                                                                 HttpMethod method,
+                                                                 Boolean returnMultipleResults) {
+        List<Pair<String, Integer>> frontends = getOtherAliveFe();
+        if (frontends.isEmpty()) {
+            return List.of();
+        }
+        ImmutableMap<String, String> header = ImmutableMap.<String, String>builder()
+                .put(HttpHeaders.AUTHORIZATION, authorization).build();
+        List<String> result = Lists.newArrayList();
+        for (Pair<String, Integer> front : frontends) {
+            String url = String.format("http://%s:%d%s", front.first, front.second, queryPath);
+            try {
+                String data = null;
+                if (method == HttpMethod.GET) {
+                    data = HttpUtils.get(url, header);
+                } else if (method == HttpMethod.POST) {
+                    data = HttpUtils.post(url, null, header);
+                }
+                if (StringUtils.isNotBlank(data)) {
+                    result.add(data);
+                }
+                if (!returnMultipleResults && !result.isEmpty()) {
+                    return result;
+                }
+            } catch (Exception e) {
+                LOG.error("request url {} error", url, e);
+            }
+        }
+        return result;
+    }
+
+    public static List<Pair<String, Integer>> getAllAliveFe() {
+
+        return GlobalStateMgr.getCurrentState()
```

Comment:
you could check if `GlobalStateMgr.getCurrentState()` is null

Reply:
Improved. Thanks !